sql/core/src/main/scala/org/apache/spark/sql/ContinuousQueryManager.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import scala.collection.mutable
+
+import org.apache.spark.annotation.Experimental
+import org.apache.spark.sql.execution.streaming.{Sink, StreamExecution}
+
+/**
+ * :: Experimental ::
+ * A class to manage all the [[org.apache.spark.sql.ContinuousQuery ContinuousQueries]] active
+ * on a [[SQLContext]].
+ * 
+ * @since 2.0.0
+ */
+@Experimental
+class ContinuousQueryManager {
+
+  private val activeQueries = new mutable.HashMap[String, ContinuousQuery]
+  private val activeQueriesLock = new Object
+  private val awaitTerminationLock = new Object
+
+  @volatile
+  private var lastTerminatedQuery: ContinuousQuery = null
+
+  /** Returns a list of active queries associated with this SQLContext */
+  def active: Array[ContinuousQuery] = activeQueriesLock.synchronized {
+    activeQueries.values.toArray
+  }
+
+  /** Returns an active query from this SQLContext or throws exception if bad name */
+  def get(name: String): ContinuousQuery = activeQueriesLock.synchronized {
+    activeQueries.get(name).getOrElse {
+      throw new IllegalArgumentException(s"There is no active query with name $name")
+    }
+  }
+
+  /**
+   * Wait until any of the queries on this SQLContext is terminated, with or without
+   * exceptions. Returns the query that has been terminated.
+   */
+  def awaitAnyTermination(): ContinuousQuery = {
+    awaitTerminationLock.synchronized {
+      lastTerminatedQuery = null
```

Comment:
I know this problem. That's the level of guarantee this API can provide, only wait on currently active queries at the time `awaitAnyTermination` is called. 

Actually I thought about this, and a part of me is saying that this API is need to be reconsidered.
- Option 1: instead of `sqlContext.awaitAnyTermination()` (where its not clear what you are waiting on), how about `sqlContext.awaitAnyTermination(query1, query2, ...)`
- Option 2: Forget this API for now. We can think about it later.